softwareComponents/shapeReconfig/src/hashing.cpp

```diff
@@ -0,0 +1,98 @@
+#include <shapeReconfig/hashing.hpp>
+
+namespace rofi::shapereconfig {
+
+using namespace rofi::configuration;
+
+size_t combineHash( size_t h1, size_t h2 ) 
+{ 
+    return h1 ^ (h2 << 1);
+}
+
+// If converted to degrees and rounded, equal degree has same hash
+size_t hashRad( float rads )
+{
+    return std::hash<int>{}( static_cast<int>( round( rads * 100 ) ) );
+}
+
+size_t hashJoint( const Joint& j )
+{
+    size_t h1 = std::accumulate( j.jointLimits().begin(), j.jointLimits().end(), 0, 
```

Comment:
You should move jointLimits to its own variable.
Otherwise the jointLimits could change to always return a different e.g. array and your code would silently stop working.

Reply:
Because if the `j.jointLimits()` creates a new iterable, you can't mix iterables from two different containers.

Since it is a span, it will have exactly the same value, because it points to the original container, but this doesn't have to be the case. (So it will work until someone changes the implementation...)